model/minio/minio_server.rb

```diff
@@ -21,6 +21,14 @@ class MinioServer < Sequel::Model
     enc.column :cert_key
   end
 
+  def generate_etc_hosts_entry
+    entries = ["127.0.0.1 #{hostname}"]
+    entries += cluster.servers.reject { _1.id == id }.map do |server|
+      "#{server.public_ipv4_address} #{server.hostname}"
```

Comment:
Wondering if we should make this conditional on provider? (not sure if we know if this problem is isolated to Hetzner setups only)

Reply:
Provider is not a factor in the poor private networking performance. Its impact is the same for all. However, I agree with @fdr , can we try if it works with IPv6?

Reply:
Updated to use public IPv6